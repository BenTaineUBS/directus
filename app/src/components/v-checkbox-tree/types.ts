export type Choice = {
	text: string;
	value: string | number;
	children?: Choice[];
};
