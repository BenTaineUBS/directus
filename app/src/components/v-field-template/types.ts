export type FieldTree = {
	field: string;
	name: string;
	key: string;
	group?: boolean;
	disabled?: boolean;
	children?: FieldTree[];
};
