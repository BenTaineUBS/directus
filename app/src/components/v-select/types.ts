export type Option = {
	value: string | number | null;
	icon?: string;
	iconColor?: string;
	text?: string;
	disabled?: boolean;
	children?: Option[];
	divider?: boolean;
	selectable?: boolean;
	hidden?: boolean;
};
