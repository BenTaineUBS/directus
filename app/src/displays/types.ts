import { DeepPartial, Field, LocalType, Type } from '@directus/shared/types';
import { Component } from 'vue';

export type DisplayHandlerFunctionContext = {
	type: string;
};

export type DisplayHandlerFunction = (
	value: any,
	options: any,
	context?: DisplayHandlerFunctionContext
) => string | null;

export type DisplayFieldsFunction = (
	options: any,
	context: {
		collection: string;
		field: string;
		type: string;
	}
) => string[];

export interface DisplayConfig {
	id: string;
	name: string;
	icon: string;
	description?: string;

	// eslint-disable-next-line @typescript-eslint/ban-types
	handler: DisplayHandlerFunction | Component | Function;
	options: null | DeepPartial<Field>[] | Component;
	types: Type[];
	groups?: LocalType[];
	fields?: string[] | DisplayFieldsFunction;
}

export type DisplayDefineParam = DisplayConfig | (() => DisplayConfig);
