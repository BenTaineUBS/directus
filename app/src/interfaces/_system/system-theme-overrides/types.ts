export interface GroupValue {
	[key: string]: string | number | undefined | GroupValue;
}

export type SetValueFn = (path: string[], value: string | number | undefined) => void;
