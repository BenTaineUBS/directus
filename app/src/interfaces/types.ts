import { Component } from 'vue';
import { Field, types, localTypes } from '@/types';

export interface InterfaceConfig {
	id: string;
	name: string;
	icon: string;
	description?: string;
	component: Component | (() => Promise<Component>);
	options: DeepPartial<Field>[] | Component | (() => Promise<Component>);
	types: typeof types[number][];
	groups?: readonly typeof localTypes[number][];
	relational?: boolean;
	hideLabel?: boolean;
	hideLoader?: boolean;
	system?: boolean;
	recommendedDisplays?: string[];
}

export type InterfaceDefineParam = InterfaceDefineParamGeneric<InterfaceConfig>;
export type InterfaceDefineParamGeneric<T> = T | (() => T);
