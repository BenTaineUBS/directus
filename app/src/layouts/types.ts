import { Component, ComponentPublicInstance } from 'vue';
import { Item } from '@/components/v-table/types';
import { Filter } from '@/types';

export interface LayoutConfig {
	id: string;
	name: string;
	icon: string;
	component: Component;
	slots: {
		options: Component;
		sidebar: Component;
		actions: Component;
	};
	setup: (LayoutOptions: LayoutProps) => any;
}

export interface LayoutProps {
	collection: string;
	selection?: Item[];
	layoutOptions: any;
	layoutQuery: any;
	filters: Filter[];
	searchQuery: string | null;
	selectMode: boolean;
	readonly: boolean;
	resetPreset?: () => Promise<void>;
}

export type LayoutContext = Record<string, any>;

export type LayoutDefineParam = LayoutConfig | ((context: LayoutContext) => LayoutConfig);

export interface LayoutComponent extends ComponentPublicInstance {
	refresh: () => Promise<void>;
}
