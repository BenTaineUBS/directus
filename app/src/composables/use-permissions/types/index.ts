import { MaybeRef } from 'vue';
import { PrimaryKey as PrimaryKeyOrig } from '@directus/types';

export type Collection = MaybeRef<string>;

export type PrimaryKey = MaybeRef<PrimaryKeyOrig | null>;

export type IsNew = MaybeRef<boolean>;
