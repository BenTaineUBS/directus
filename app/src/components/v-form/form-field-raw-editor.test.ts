import { it, test, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';

import formFieldRawEditor from './form-field-raw-editor.vue';
import { GlobalMountOptions } from '@vue/test-utils/dist/types';

const i18n = createI18n();

const global: GlobalMountOptions = {
	plugins: [i18n],
};

test('should render', () => {
	expect(formFieldRawEditor).toBeTruthy();

	const wrapper = mount(formFieldRawEditor, {
		props: {
			showModal: true,
			field: 'object',
			disabled: false,
			currentValue: '["id","new_content"]',
		},
		global,
	});
	expect(wrapper.html()).toMatchSnapshot();
});

// test if there is a value
test('submitting', async () => {
	expect(formFieldRawEditor).toBeTruthy();
	const wrapper = mount(formFieldRawEditor, {
		props: {
			showModal: false,
			field: 'string',
			disabled: false,
			currentValue: 'things',
		},
		global,
	});
	await wrapper.find('v-button').trigger('click');
	await wrapper.vm.$nextTick();
	expect(wrapper.emitted().submit.length).toBe(1);
});

it('should cancel', async () => {
	const wrapper = mount(formFieldRawEditor, {
		props: {
			showModal: true,
			field: 'object',
			disabled: false,
			currentValue: '["id","new_content"]',
		},
		global,
	});
	await wrapper.trigger('keydown.esc');
	await wrapper.vm.$nextTick();
	expect(wrapper.emitted().cancel).toBeTruthy;
});
