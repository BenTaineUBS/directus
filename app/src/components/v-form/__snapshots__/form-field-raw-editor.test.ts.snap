// Vitest Snapshot v1

exports[`click on done 1`] = `
"<v-dialog model-value=\\"true\\" persistent=\\"\\">
  <v-card>
    <v-card-title>edit_raw_value</v-card-title>
    <v-card-text>
      <interface-input-code disabled=\\"false\\" class=\\"raw-value\\" placeholder=\\"enter_raw_value\\" language=\\"plaintext\\"></interface-input-code>
    </v-card-text>
    <v-card-actions>
      <v-button>done</v-button>
    </v-card-actions>
  </v-card>
</v-dialog>"
`;

exports[`form-field-raw-editor.vue 1`] = `
"<v-dialog model-value=\\"true\\" persistent=\\"\\">
  <v-card>
    <v-card-title>edit_raw_value</v-card-title>
    <v-card-text>
      <interface-input-code disabled=\\"false\\" class=\\"raw-value\\" placeholder=\\"enter_raw_value\\" language=\\"plaintext\\"></interface-input-code>
    </v-card-text>
    <v-card-actions>
      <v-button>done</v-button>
    </v-card-actions>
  </v-card>
</v-dialog>"
`;

exports[`should render 1`] = `
"<v-dialog model-value=\\"true\\" persistent=\\"\\">
  <v-card>
    <v-card-title>edit_raw_value</v-card-title>
    <v-card-text>
      <interface-input-code disabled=\\"false\\" class=\\"raw-value\\" placeholder=\\"enter_raw_value\\" language=\\"plaintext\\"></interface-input-code>
    </v-card-text>
    <v-card-actions>
      <v-button>done</v-button>
    </v-card-actions>
  </v-card>
</v-dialog>"
`;
